### PR TITLE
New random numbers

### PR DESCRIPTION
**Pull request by martinduke**
As requested, all new version number, salts, etc. for QUICv2.

Verification of the test vectors would be especially valuable.

### CODE REVIEW

File: draft-ietf-quic-v2.md [hunk 1]
```diff
@@ -91,8 +91,8 @@ version 1 specification as described in {{QUIC}}, {{QUIC-TLS}}, and
 
 ## Version Field
 
-The Version field of long headers is 0x709a50c4. This randomly chosen value is
-meant to avoid creating patterns in specified version numbers.
+The Version field of long headers is 0x5a8e301a. This was generated by taking
+the first four bytes of the sha256sum of "QUICv2 version number."
```
Comment (martinthomson) [inline draft-ietf-quic-v2.md]: Took me a while to work this out:

```
$ echo -n "QUICv2 version number." | sha256sum
43027d66a59ea521a5f2e5281505c0ae10ad363cded1053de316415f18b1b2e0  -
~
$ echo -n "QUICv2 version number" | sha256sum
6b3343cfc25b51101996471c75a1de7b3d4da11ad87f165d3968261e230b96ff  -
~
$ echo  "QUICv2 version number" | sha256sum
5a8e301ab1a95e157c6f303e47b89e9f762db8e44e342628b5f309a5b943d5ae  -
```

You might want to look into that.

File: draft-ietf-quic-v2.md [hunk 2]
```diff
@@ -118,9 +118,11 @@ The values below were chosen randomly.
 The salt used to derive Initial keys in {{Section 5.2 of QUIC-TLS}} changes to:
 
 ~~~
-initial_salt = 0xa707c203a59b47184a1d62ca570406ea7ae3e5d3
+initial_salt = 0x3a785fb8f3009837546c9e8570b6bac843e3e142
 ~~~
 
+This is the first 20 bytes of the sha256sum of "QUICv2 salt."
```
Comment (martinthomson) [inline draft-ietf-quic-v2.md]: Same as above.  Note that this assumes that echo is producing '\n' and not '\r\n' or any of the alternatives, so you might want to use `echo -n` instead.  Also, the period inside the quotes is definitely misleading.

File: draft-ietf-quic-v2.md [hunk 3]
```diff
@@ -143,11 +145,15 @@ change to:
...
 secret =
-  0x3425c20cf88779df2ff71e8abfa78249891e763bbed2f13c048343d348c060e2
-key = 0xba858dc7b43de5dbf87617ff4ab253db
-nonce = 0x141b99c239b03e785d6a2e9f
+  0x66e90fdd03db030d5c80eda22a45456cf664e37f719251105953350726a609cd
+key = 0xd42ce594316b516e9db972acf7656505
+nonce = 0xac3aac3139c1f33b5c9b6091
 ~~~
 
+The secret is the sha256sum of "QUICv2 Retry Secret". The key and nonce are
```
Comment (martinthomson) [inline draft-ietf-quic-v2.md]: This is worse again than the previous ones, aside from the moved period.  I had to knock the capitals off to get the value you have.

**Review by martinthomson — COMMENTED** [review]
A few troubles reproducing what you have there.  I'll stop at the invocations of SHA-256.

**Comment by martinduke** [comment]
Woof. Thank you very much for quickly checking, and sorry I botched every part of that.

All fixed now, I hope.

File: draft-ietf-quic-v2.md [hunk 4]
```diff
@@ -143,11 +145,15 @@ change to:
 
 ~~~
 secret =
-  0x3425c20cf88779df2ff71e8abfa78249891e763bbed2f13c048343d348c060e2
-key = 0xba858dc7b43de5dbf87617ff4ab253db
-nonce = 0x141b99c239b03e785d6a2e9f
+  0xc4dd2484d681aefa4ff4d69c2c20299984a765a5d3c31982f38fc74162155e9f
+key = 0x8fb4b01b56ac48e260fbcbcead7ccc92
+nonce = 0xd86969bc2d7c6d9990efb04a
```
Comment (martinthomson) [inline draft-ietf-quic-v2.md:150]: Checks out now.

**Review by martinthomson — APPROVED** [review]
WFM.  Not validated with a real build, but a script.  See https://github.com/quicwg/base-drafts/pull/4997 for the script that I used.

**Comment by kazu-yamamoto** [comment]
The new test vectors works well with my QUIC implementation in Haskell.

File: draft-ietf-quic-v2.md (hunk 4 above)
Comment (martinduke) [reply draft-ietf-quic-v2.md:150]: Thanks @martinthomson!

**Comment by martinduke** [comment]
> The new test vectors works well with my QUIC implementation in Haskell.

Thanks Kazu!

**Pull request merged**